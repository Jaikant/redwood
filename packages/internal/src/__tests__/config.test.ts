import MockProject from '@redwoodjs/test-mocks'

import {
  getConfig,
  DEFAULT_CONFIG,
  getConfigSides,
  getSideConfig,
} from '../config'

describe('config', () => {
  const mockrw = new MockProject()

  beforeEach(() => {
    mockrw.mock()
  })

  afterAll(() => {
    mockrw.restore()
  })

  describe('getConfig', () => {
    it('returns a default config', () => {
      expect(getConfig()).toEqual(DEFAULT_CONFIG)
    })

    it('the default config is merged with `redwood.toml`', () => {
      mockrw.setPaths((paths) => {
        paths['redwood.toml'] = `
          [web]
            port = 8080
          [api]
            port = 8081
        `
        return paths
      })
      const config = getConfig()
      expect(config.web.port).toEqual(8080)
      expect(config.api.port).toEqual(8081)
    })
  })

  describe('getConfigSides', () => {
    it('returns the config sides', () => {
      const sides = getConfigSides()
      mockrw.restore()
      expect(sides).toMatchSnapshot()
    })
  })

  describe('getSideConfig', () => {
    it('returns the correct side', () => {
      expect(getSideConfig('web').name).toEqual('web')
    })

    it('throws when you try to get a config that does not exist', () => {
      expect(() => getSideConfig('rambo')).toThrow()
    })
  })
})
