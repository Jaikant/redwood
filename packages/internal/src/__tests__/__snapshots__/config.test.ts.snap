// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config getConfigSides returns the config sides 1`] = `
Object {
  "api": Object {
    "build": Array [
      Object {
        "command": "NODE_ENV=production babel src --out-dir dist",
        "destination": "./dist",
        "name": "default",
        "source": "./src",
      },
    ],
    "host": "localhost",
    "name": "api",
    "path": "./api",
    "port": 8081,
    "target": "node",
  },
  "web": Object {
    "apiProxyPath": "/.netlify/functions",
    "apiProxyPort": 8911,
    "build": Array [
      Object {
        "command": "yarn webpack --config ../node_modules/@redwoodjs/core/config/webpack.production.js",
        "destination": "./dist",
        "name": "default",
        "source": "./src",
      },
      Object {
        "command": "yarn webpack --config ../node_modules/@redwoodjs/core/config/webpack.stats.js",
        "destination": "./dist",
        "name": "stats",
        "source": "./src",
      },
    ],
    "host": "localhost",
    "name": "web",
    "path": "./web",
    "port": 8080,
    "target": "browser",
  },
}
`;
